modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/bean/APICreationRequestBean.java

```diff
@@ -0,0 +1,342 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.am.integration.test.utils.bean;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.wso2.am.integration.test.utils.APIManagerIntegrationTestException;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Bean class to contain the Information needed to create and update process of a API.
+ */
+public class APICreationRequestBean extends AbstractRequest {
+
+    private static final Log log = LogFactory.getLog(APIRequest.class);
+    private String name;
+    private String context;
+    private JSONObject endpoint;
+    private String version;
+    private String visibility = "public";
+    private String description = "description";
+    private String endpointType = "nonsecured";
+    private String http_checked = "http";
+    private String https_checked = "https";
+    private String tags = "tags";
+    private String tier = "Silver";
+    private String thumbUrl = "";
+    private String tiersCollection = "Gold";
+    private String resourceCount = "0";
+    private String roles = "";
+    private String wsdl = "";
+    private String default_version = "";
+    private String default_version_checked = "";
+    private List<APIResourceBean> resourceBeanList;
+    private String epUsername = "";
+    private String epPassword = "";
+    private String sandbox = "";
+    private String provider = "admin";
+
+
+    /**
+     * constructor of the APICreationRequestBean class
+     *
+     * @param apiName     - Name of the APi
+     * @param context     - API context
+     * @param version     - API version
+     * @param endpointUrl - Endpoint URL of the API
+     * @throws APIManagerIntegrationTestException
+     */
+    public APICreationRequestBean(String apiName, String context, String version, URL endpointUrl) throws APIManagerIntegrationTestException {
+        this.name = apiName;
+        this.context = context;
+        this.version = version;
+        resourceBeanList = new ArrayList<APIResourceBean>();
+        resourceBeanList.add(new APIResourceBean("GET", "Application & Application User", "Unlimited", "/*"));
+        try {
+            this.endpoint = new JSONObject("{\"production_endpoints\":{\"url\":\""
+                    + endpointUrl + "\",\"config\":null},\"endpoint_type\":\""
+                    + endpointUrl.getProtocol() + "\"}");
+        } catch (JSONException e) {
+            log.error("JSON construct error", e);
+            throw new APIManagerIntegrationTestException(" Error When constructing the edn point url JSON", e);
+        }
+
+
+    }
+
+    @Override
```

Comment:
No need to have doc comments for overridden methods. java doc generation will inherit the super class method doc comment if its not available in subclass.